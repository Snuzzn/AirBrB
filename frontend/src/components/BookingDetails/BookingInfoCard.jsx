import React from 'react'

import PropTypes from 'prop-types';

function BookingInfoCard ({ title, message, icon, bg }) {
  return (
    <div className="flex flex-col h-96 lg:h-36 lg:flex-row w-full gap-5">
      <div className="bg-white  rounded-lg shadow flex-1 flex justify-start">
        <div className="flex gap-6 items-center ml-9">
          <div className={bg + ' p-5 rounded-2xl'} id="bookingMetricIcon">
            {icon}
          </div>
          <div className="">
            <p className="text-gray-500 font-light text-sm" id="bookingMetricText">{title}</p>
            <p className="text-xl font-bold" id="bookingMetricMessage">{message}</p>
          </div>
        </div>
      </div>
    </div>
  )
}

export default BookingInfoCard

BookingInfoCard.propTypes = {
  title: PropTypes.string,
  message: PropTypes.string,
  icon: PropTypes.node,
  bg: PropTypes.string
}
