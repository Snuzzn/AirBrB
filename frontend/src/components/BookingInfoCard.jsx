import React from 'react'

import PropTypes from 'prop-types';

function BookingInfoCard ({ message, icon, bg }) {
  return (
    <div className="flex flex-col h-96 lg:h-36 lg:flex-row w-full gap-5">
      <div className="bg-white  rounded-lg shadow flex-1 flex justify-start">
        <div className="flex gap-6 items-center ml-9">
          <div className={bg + ' p-5 rounded-2xl'}>
            {icon}
          </div>
          <div className="">
            <p className="text-gray-500 font-light text-sm">Listing Age</p>
            <p className="text-xl font-bold">{message}</p>
          </div>
        </div>
      </div>
    </div>
  )
}

export default BookingInfoCard

BookingInfoCard.propTypes = {
  message: PropTypes.string,
  icon: PropTypes.node,
  bg: PropTypes.string
}
