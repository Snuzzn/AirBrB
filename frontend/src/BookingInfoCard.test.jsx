import React from 'react';
import { shallow } from 'enzyme';
import BookingInfoCard from './components/BookingDetails/BookingInfoCard';
import { HiCake } from 'react-icons/hi'
import renderer from 'react-test-renderer'

describe('BookingInfoCard', () => {
  // const noop = () => {};
  /**
   * BookingInfoCard takes title, message, icon, and background colour to
   * display the appropriate metric. It is used in the Booking Details page.
   */

  it('renders title', () => {
    const card = shallow(<BookingInfoCard title='$30'/>)
    const title = card.find('#bookingMetricText').text()
    expect(title).toEqual('$30')
  })

  it('renders message', () => {
    const card = shallow(<BookingInfoCard message='Yearly Profit'/>)
    const message = card.find('#bookingMetricMessage').text()
    expect(message).toEqual('Yearly Profit')
  })

  it('renders icon', () => {
    const card = shallow(<BookingInfoCard icon={<HiCake/>}/>)
    const icon = card.find('#bookingMetricIcon')
    const inputIcon = <HiCake/>
    expect(icon.contains(inputIcon)).toEqual(true)
  })

  it('applies appropriate background colour around icon', () => {
    const card = shallow(<BookingInfoCard bg="bg-red-200"/>)
    const icon = card.find('#bookingMetricIcon')
    expect(icon.hasClass('bg-red-200')).toEqual(true)
  })

  // Snapshots
  it('renders with minimal props', () => {
    const card = renderer.create(<BookingInfoCard/>).toJSON();
    expect(card).toMatchSnapshot();
  })

  it('renders with provided data', () => {
    const card = renderer.create(<BookingInfoCard
      title='$30'
      message='Yearly Profit'
      icon={<HiCake/>}
      bg="bg-red-200"
    />).toJSON();
    expect(card).toMatchSnapshot();
  })
})
