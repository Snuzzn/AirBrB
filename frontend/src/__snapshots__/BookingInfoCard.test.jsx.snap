// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingInfoCard renders with minimal props 1`] = `
<div
  className="flex flex-col h-96 lg:h-36 lg:flex-row w-full gap-5"
>
  <div
    className="bg-white  rounded-lg shadow flex-1 flex justify-start"
  >
    <div
      className="flex gap-6 items-center ml-9"
    >
      <div
        className="undefined p-5 rounded-2xl"
        id="bookingMetricIcon"
      />
      <div
        className=""
      >
        <p
          className="text-gray-500 font-light text-sm"
          id="bookingMetricText"
        />
        <p
          className="text-xl font-bold"
          id="bookingMetricMessage"
        />
      </div>
    </div>
  </div>
</div>
`;
